connectors/cloud-storage-sink/2.5.1/cloud-storage-sink.md

```diff
@@ -0,0 +1,208 @@
+
+## Features
+The Pulsar IO Cloud Storage Sink connector provides the following features:
+
+- Exactly Once Delivery: Records that are exported using a deterministic partitioner are delivered with exactly-once semantics regardless of the eventual consistency of Cloud Storage.
+- Data Format with or without a Schema: Plug and play, the connector supports writing data to cloud storage in Avro, JSON and parquet formats. Generally, the connector can accept any format that provides an implementation of the Format interface.
+- Partitioner: The connector supports the TimeBasedPartitioner class based on the Pulsar message publishTime TimeStamp. Time-based partitioning options are daily or hourly.
+- More object storage support: The connector uses jclouds as an implementation of cloud storage. You can use the jclouds object storage jar to connect to more types of object storage. If you need to customize credentials, you can register ʻorg.apache.pulsar.io.jcloud.credential.JcloudsCredential` via SPI.
+
+The Cloud Storage sink connector pulls messages from Pulsar topics and persists messages to Cloud Storage.
+
+# Installation
+
+```
+git clone https://github.com/streamnative/pulsar-io-cloud-storage.git
+cd pulsar-io-cloud-storage/
+mvn clean install -DskipTests
+cp target/pulsar-io-cloud-storage-0.0.1.nar $PULSAR_HOME/pulsar-io-cloud-storage-0.0.1.nar
+```
+
+# Configuration 
+
+The Cloud Storage sink connector supports the following properties.
+
+## Cloud Storage sink connector configuration
+
+| Name | Type|Required | Default | Description |
+|------|----------|----------|---------|-------------|
+| `provider` |String| True | " " (empty string) | The Cloud Storage type. for example, `aws-s3`,`gcs`|
+| `accessKeyId` |String| True | " " (empty string) | The Cloud Storage access key ID. |
+| `secretAccessKey` | String| True | " " (empty string) | The Cloud Storage secret access key. |
+| `role` | String |False | " " (empty string) | The Cloud Storage role. |
+| `roleSessionName` | String| False | " " (empty string) | The Cloud Storage role session name. |
+| `bucket` | String| True | " " (empty string) | The Cloud Storage bucket. |
+| `endpoint` | String| False | " " (empty string) | The Cloud Storage endpoint. |
+| `formatType` | String| False | "json" | The data format type: JSON, Avro, or Parquet. By default, it is set to JSON. |
+| `partitionerType` | String| False |"partition" | The partition type. It can be configured by partition or time. By default, the partition type is configured by partition. |
```

Comment:
```suggestion
| `partitionerType` | String| False |"partition" | The partitioning type. It can be configured by topic partitions or by time. By default, the partition type is configured by topic partitions. |
```